### PR TITLE
refactor(#2)

### LINKED ISSUE #2: Refactor Renewer and Storage for Easier integration

### PR DESCRIPTION
**Pull request by ShivanshVij**
This PR closes #2 and #1 and achieves the following:

- Refactoring the renewer package into `pkg/acme` package
- Refactoring certifier's DNS server into its own `pkg/dns` package
- Adding test cases for the `pkg/dns` package
- Adding comments and explanations (godoc style) for everything in Certifier
- moving `utils` package into the `pkg` directory
- moving the `memory` implementation of `storage.Storage` into the `internal` directory
- moving the `options` struct into its own `pkg/options` package

### CODE REVIEW

File: pkg/acme/provider.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+import (
+	"github.com/go-acme/lego/v4/challenge"
+	"github.com/go-acme/lego/v4/challenge/dns01"
+	"github.com/loopholelabs/certifier/pkg/options"
+	"github.com/loopholelabs/certifier/pkg/storage"
+	"github.com/loopholelabs/logging"
+)
+
+var _ challenge.Provider = (*provider)(nil)
```
Comment (SuperManifolds) [inline pkg/acme/provider.go:11]: Could you explain what is going on here?

File: pkg/dns/dns.go [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+}
+
+// defaultSOA returns an SOA Record with the defaults filled in
+func (d *DNS) defaultSOA(domain string) *dns.SOA {
+	return &dns.SOA{
+		Hdr: dns.RR_Header{
+			Name:   dns.Fqdn(domain),
+			Rrtype: dns.TypeSOA,
+			Class:  dns.ClassINET,
+			Ttl:    86400,
```
Comment (SuperManifolds) [inline pkg/dns/dns.go]: Might be worth making these numerical values into configurable constants at the top of the file

File: pkg/dns/dns_test.go [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+	const domain = "default.domain"
+
+	d := New(rootDomain, publicDomain)
+
+	soa := d.defaultSOA(domain)
+	assert.Equal(t, dns.Fqdn(domain), soa.Hdr.Name)
+	assert.Equal(t, uint16(dns.ClassINET), soa.Hdr.Class)
+	assert.Equal(t, dns.TypeSOA, soa.Hdr.Rrtype)
+	assert.Equal(t, uint32(86400), soa.Hdr.Ttl)
```
Comment (SuperManifolds) [inline pkg/dns/dns_test.go]: Same comment re: constants

File: pkg/acme/provider.go (hunk 1 above)
Comment (ShivanshVij) [reply pkg/acme/provider.go:11]: This line creates an empty variable that tells the compiler that *provider should fulfill the challenge.Provider interface.

It's basically a compiler check that we are fulfilling the interface

**Pull request merged**